### PR TITLE
Umbrella conf support for distillery plugin

### PR DESCRIPTION
**Pull request by coryodaniel**
* Backported elixir 1.4 Mix.Project.apps_paths/1
* Concat config files
* Merge schema files

### CODE REVIEW

**Comment by bitwalker** [comment]
> Im assuming since we are concat and merging the conf and schema from conf and schema files people are already editing, the actual conf/schema files for the umbrella should probably not be actual files in the repo, probably just made in memory and spit out in the release folder. @bitwalker, thoughts?

The latter. We'd want to generate the conf/schema in the release output, but it should be assembled in memory and output there as a generated artifact, rather than something users maintain. At least that's my gut feeling on it right now.

**Comment by coryodaniel** [comment]
Ok, looks like this is working! See this umbrellified version of your [test app](https://github.com/coryodaniel/distillery-umbrella-test). Its pointing at my conform branch for the fixes.

**Comment by coryodaniel** [comment]
There are some debugging outputs in the sh file so you can see its loading the correct paths when you start your app... I'll pull those after you take a peek!

**Comment by coryodaniel** [comment]
@bitwalker Looks like this is working! See this umbrellified version of your [test app](https://github.com/coryodaniel/distillery-umbrella-test). Its pointing at my conform branch for the fixes.

There are some debugging outputs in the `priv/bin/pre_start.sh` so you can see its loading the correct paths when you start your app... I'll delete those after you take a peek!

**Comment by coryodaniel** [comment]
@bitwalker have you had a chance to check this out?

File: priv/bin/pre_start.sh [hunk 1]
```diff
@@ -10,6 +10,9 @@ if [ -z "$CONFORM_SCHEMA_PATH" ]; then
     fi
 fi
 
+echo "[DEBUG] schema path"
```
Comment (bitwalker) [inline priv/bin/pre_start.sh]: Can we remove this, or put it under a check for the presence of the `DEBUG_BOOT` env var which is set by the release boot script? I'd rather not have extraneous output unless it's necessary.

**Comment by bitwalker** [comment]
Sorry for the delay, just one comment and I'll get this merged :)

**Comment by coryodaniel** [comment]
No prob! All clear!

**Comment by bitwalker** [comment]
Thanks!

**Pull request merged**